sdk/identity/identity/src/credentials/managedIdentityCredential/index.ts

```diff
@@ -196,7 +196,20 @@ export class ManagedIdentityCredential implements TokenCredential {
         message: err.message
       });
 
+      // If either the network is unreachable,
+      // we can safely assume the credential is unavailable.
       if (err.code === "ENETUNREACH") {
+        const error = new CredentialUnavailable(
+          "ManagedIdentityCredential is unavailable. Network unreachable."
+        );
+
+        logger.getToken.info(formatError(scopes, error));
+        throw error;
+      }
+
+      // If either the host was unreachable,
+      // we can safely assume the credential is unavailable.
+      if (err.code === "EHOSTUNREACH") {
```

Comment:
Do these map to unique `statusCode` values so we can stay consistent with the code below? Just curious!

Reply:
I couldn't find any meaningful map between these codes and a statusCode in Node's source. Later if we find a clear map, we can clean this up 🌞